src/tactic/compute_degree.lean

```diff
@@ -127,23 +185,91 @@ meta def eval_guessing (n : ℕ) : expr → tactic ℕ
 | `(max %%a %%b) := max <$> eval_guessing a <*> eval_guessing b
 | e              := eval_expr' ℕ e <|> pure n
 
-/--  A general description of `compute_degree_le_aux` is in the doc-string of `compute_degree`.
-The difference between the two is that `compute_degree_le_aux` makes no effort to close side-goals,
-nor fails if the goal does not change. -/
-meta def compute_degree_le_aux : tactic unit := do
-try $ refine ``(degree_le_nat_degree.trans (with_bot.coe_le_coe.mpr _)),
-`(nat_degree %%tl ≤ %%tr) ← target |
-  fail "Goal is not of the form\n`f.nat_degree ≤ d` or `f.degree ≤ d`",
-expected_deg ← guess_degree tl >>= eval_guessing 0,
-deg_bound ← eval_expr' ℕ tr <|> pure expected_deg,
-if deg_bound < expected_deg
-then fail sformat!"the given polynomial has a term of expected degree\nat least '{expected_deg}'"
-else repeat $ resolve_sum_step
+/--  These are the cases in which an easy lemma computes the degree. -/
```

Comment:
```suggestion
/-- If an easy lemma computes the degree in the goal, fail and suggest that lemma. -/
```

Reply:
The suggestions lemma has been revamped: let me know if you like it!